packages/cli/src/lib/setup/setupUpgrade.ts

```diff
@@ -221,21 +238,28 @@ function Upgrade(options) {
                     for (const instance of locInstances) {
                         try {
                             if (
-                                !semver.satisfies(instance.value.common.version, deps[dName], {
+                                // @ts-expect-error InstaceCommon has version: TODO fix types
+                                !semver.satisfies(instance.value!.common.version, deps[dName], {
```

Comment:
Are we sure that `instance.value` is always defined? If so, is some type wrong, allowing undefined?

Reply:
FWIW, I'd put this at the start of the entire block
```ts
// @ts-expect-error InstaceCommon has version: TODO fix types
const instanceVersion: string = instance.value!.common.version;
```
and then just use `instanceVersion` below.

Reply:
> Yes, according to TS `InstanceCommon` has no version. Of course it has.

I didn't ask about `common`. `instance.value` may be undefined, but you put `!` after each usage, meaning it probably isn't undefined?



> The blocks themselves are very short and only one usage is there, thus I don't really see the benefit here.

I count 3 usages and 3 identical `// @ts-expect-error`s per `for` loop. That would be enough for me to avoid repetition:
![grafik](https://user-images.githubusercontent.com/17641229/187243244-b6ce8f26-c20c-4584-a7b6-5af83576e6a1.png)

Reply:
Maybe I have looked in the wrong spot, I will look again later